client/src/research/InteractionsView.js

```diff
@@ -1,13 +1,34 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import {InteractionTypes} from '../shared/data.js';
-
-
+import { VictoryBar, VictoryChart, VictoryTheme, VictoryGroup, VictoryAxis, VictoryLabel} from 'victory'; 
+import _ from 'lodash';
 // Render a list of logged user interactions
 class InteractionsView extends Component {
-  render() {  
-    // unpack!
-    //filter out testing data
+  totalSwipes(interactions, key){
+    return _.countBy(interactions, row => row.interaction.turn[key]);
+  }
+
+  percentRightPerProfile(interactions, key){
+    const numSwipes = this.totalSwipes(interactions, key);
+    var totals = {};
+    var percents = {};
```

Comment:
The variable names here are a bit confusing (`numSwipes`, `totals` and `percents`).

Is there a way to create the "percent map" in only one loop?  So that you can get the "totals map" by calling `totalSwipes` up top, and then the loop builds up the percent?  I think that could help simplify what's happening in this function.